app/src/main/java/com/sopt/now/presentation/auth/login/LoginActivity.kt

```diff
@@ -34,19 +40,16 @@ class LoginActivity : AppCompatActivity() {
     }
 
     private fun observeLogin() {
-        loginViewModel.status.observe(this) {
-            if (it) {
-                navigateToMain()
-                showToast(
-                    ApplicationClass.SharedPreferences.sSharedPreferences.getString(
-                        MEMBER_ID,
-                        null
-                    ) ?: ""
-                )
-            } else {
-                showToast(loginViewModel.errorMessage ?: "")
+        loginViewModel.state.flowWithLifecycle(lifecycle).onEach { state ->
+            when (state) {
+                is UiState.LOADING -> {}
+                is UiState.SUCCESS<*> -> {
```

Comment:
```suggestion
                is UiState.LOADING -> Unit
                is UiState.SUCCESS -> {
```
이렇게도 사용가능합니다!

추가로 state 와 flow를 생명주기를 고려하여 onEach로  정확하게 하셨네요!